larq_compute_engine/mlir/transforms/optimize.cc

```diff
@@ -97,37 +97,43 @@ struct SetBconvReadWriteBitpacked : public OpRewritePattern<TF::LceBconv2dOp> {
             : dyn_cast_or_null<TF::LceBconv2dOp>(
                   outer_bconv_op.input().getDefiningOp());
 
-    if (!inner_bconv_op) return failure();
-
-    if (inner_bconv_op.padding() == "SAME" && inner_bconv_op.pad_values() != 1)
+    if (inner_bconv_op && inner_bconv_op.padding() == "SAME" &&
+        inner_bconv_op.pad_values() != 1)
       return failure();
```

Comment:
I don't yet understand this part of the code, but in case there is an `inner_bconv_op` but it has the wrong padding, then we should still use binarymaxpool right? But simply not with int32 input. I would code it like this:
```suggestion
    if (inner_bconv_op && inner_bconv_op.padding() == "SAME" &&
        inner_bconv_op.pad_values() != 1)
      inner_bconv_op = nullptr;
```

Reply:
Ah yes good point, I'll fix that